keras/distribution/distribution_lib.py

```diff
@@ -507,6 +529,28 @@ def _maybe_populate_device_mesh(self, layout):
 LayoutMap.get.__doc__ = LayoutMap.__getitem__.__doc__
 
 
+@keras_export("keras.distribution.relayout")
+def relayout(value, tensor_layout):
+    """Change the layout of a Tensor value in the jit function execution.
+
+    Note that this might not work outside of the jitted function for certain
+    backend. To change the layout of a value eagerly, please use
+    `backend.distribution_lib.distribute_value`.
+
+    Args:
+        value: a Tensor to change the layout.
+        tensor_layout: TensorLayout to be applied on the value.
+
+    Returns:
+        a new value with the specified tensor layout.
+    """
+    if isinstance(value, KerasTensor):
+        # keras tensor is only used for building functional model, and can't be
+        # used to alter layout/sharding.
+        return value
```

Comment:
Great question. The issue I hit when using kerasTensor with the jax sharding API is that it will always try to convert the KerasTensor to jax array, which result into error. It might make sense to add layout only when KerasTensor is a subclass or jax array or tf.Tensor.